devmand/gateway/src/devmand/channels/cli/SshSessionAsync.cpp

```diff
@@ -66,9 +66,12 @@ Future<Unit> SshSessionAsync::write(const string& command) {
 }
 
 Future<string> SshSessionAsync::readUntilOutput(const string& lastOutput) {
-  return via(executor.get(), [this, lastOutput] {
-    return this->readUntilOutputBlocking(lastOutput);
-  });
+  this->readingState.currentLastOutput = lastOutput;
```

Comment:
Do we reset currentLastOutput properly when we finish reading until output ?

Reply:
No we do not. It should not be a problem though. 

Note: I will have to add a synchronization boolean "matchingExpectedOutput" for correctness :-(